colossalai/shardformer/layer/loss.py

```diff
@@ -86,7 +86,10 @@ def forward(
         dist.all_reduce(pred_logits, op=dist.ReduceOp.SUM, group=process_group)
         exp_logits = vocab_logits
         torch.exp(vocab_logits, out=exp_logits)
-        sum_exp_logits = torch.sum(exp_logits, dim=-1)
+        if exp_logits.dtype == torch.float16:
```

Comment:
Can you add `dtype` to arguments, like https://pytorch.org/docs/stable/generated/torch.nn.functional.softmax.html#torch.nn.functional.softmax

Reply:
Finish it in [commit](https://github.com/hpcaitech/ColossalAI/pull/5702/commits/ca4b96ab6c48d9d87276a6256fc9b8f55e97aac0).